app/lib/efile/az/az301_calculator.rb

```diff
@@ -0,0 +1,83 @@
+module Efile
+  module Az
+    class Az301Calculator < ::Efile::TaxCalculator
+      attr_reader :lines, :value_access_tracker
+
+      def initialize(value_access_tracker:, lines:, intake:)
+        @value_access_tracker = value_access_tracker
+        @lines = lines
+        @intake = intake
+      end
+
+      def calculate
+        set_line(:AZ301_LINE_6a, method(:calculate_line_6a))
+        set_line(:AZ301_LINE_6c, method(:calculate_line_6c))
+        set_line(:AZ301_LINE_7a, method(:calculate_line_7a))
+        set_line(:AZ301_LINE_7c, method(:calculate_line_7c))
+        set_line(:AZ301_LINE_26, method(:calculate_line_26))
+        set_line(:AZ301_LINE_27, method(:calculate_line_27))
+        set_line(:AZ301_LINE_32, method(:calculate_line_32))
+        set_line(:AZ301_LINE_33, method(:calculate_line_33))
+        set_line(:AZ301_LINE_34, method(:calculate_line_34))
+        set_line(:AZ301_LINE_40, method(:calculate_line_40))
+        set_line(:AZ301_LINE_41, method(:calculate_line_41))
+        set_line(:AZ301_LINE_60, method(:calculate_line_60))
+        set_line(:AZ301_LINE_62, method(:calculate_line_62))
+      end
+
+      private
+
+      def calculate_line_6a # Credit for Contributions to Qualifying Charitable Organizations from current year
+        @lines[:AZ321_LINE_20]&.value
+      end
+
+      def calculate_line_6c # Total credit for Contributions to Qualifying Charitable Organizations
+        @lines[:AZ321_LINE_22]&.value
+      end
+
+      def calculate_line_7a # Credit for Contributions Made or Fees Paid to Public Schools from current year
+        @lines[:AZ322_LINE_20]&.value
+      end
+
+      def calculate_line_7c # Total credit for Contributions Made or Fees Paid to Public Schools
+        @lines[:AZ322_LINE_22]&.value
+      end
+
+      def calculate_line_26
+        (1..24).sum { |line_num| line_or_zero("AZ301_LINE_#{line_num}c") }
```

Comment:
qq: this is maybe outside the domain of code review but i see the subtask says add lines 1-24: is that the same as adding 1c-24c?

Reply:
yeah! the c columns for each line is the total